src/psbt/mod.rs

```diff
@@ -24,16 +24,17 @@ impl PsbtUtils for PSBT {
             return None;
         }
 
-        if let Some(input) = self.inputs.get(input_index) {
-            if let Some(wit_utxo) = &input.witness_utxo {
-                Some(wit_utxo.clone())
-            } else if let Some(in_tx) = &input.non_witness_utxo {
-                Some(in_tx.output[tx.input[input_index].previous_output.vout as usize].clone())
-            } else {
-                None
+        match self.inputs.get(input_index) {
```

Comment:
I do agree this code looks less readable than before, if the specific lint is not usable maybe `#[allow(clippy::all)]` with a comment for this block?